### PR TITLE
Connector Stuck on Incorrect Container Names

### PR DESCRIPTION
**Pull request by tvaron3**
# Description
Source and sink connectors will now fail with exception message if try to create one with incorrect container names. Also fixed powershell script for building jars for tests.  Moved docker images to an azure container repository for preventing rate limiting from dockerhub.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Connector status appears as `FAILED` if try to create a connector with an incorrect container names - See [PR 41160](https://github.com/Azure/azure-sdk-for-java/pull/41160)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md:10]:
```suggestion
* Connector status appears as `FAILED` if try to create a connector with incorrect container names - See [PR 41160](https://github.com/Azure/azure-sdk-for-java/pull/41160) 
```

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/MetadataMonitorThread.java [hunk 2]
```diff
@@ -70,6 +71,10 @@ public MetadataMonitorThread(
         this.metadataReader = metadataReader;
         this.cosmosClient = cosmosClient;
         this.containersQuerySpec = this.getContainersQuerySpec();
+        int numContainers = Objects.requireNonNull(this.getAllContainers().block()).size();
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/MetadataMonitorThread.java:74]: to be consistent with sink connector, can we also have this check in the source connector startup method vs here?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-kafka-connect/src/test/connectorPlugins/build.ps1 [hunk 3]
```diff
@@ -1,20 +1,21 @@
 #!/usr/bin/env pwsh
 $ErrorActionPreference='Stop'
-cd $PSScriptRoot
+Set-Location $PSScriptRoot
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-kafka-connect/src/test/connectorPlugins/build.ps1:3]: Thanks for fixing the script 👍

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/MetadataMonitorThread.java (hunk 2 above)
Comment (tvaron3) [reply sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/MetadataMonitorThread.java]: moved it to be consistent with the sink connector.

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSinkConnector.java [hunk 4]
```diff
@@ -41,6 +47,37 @@ public void start(Map<String, String> props) {
         LOGGER.info("Starting the kafka cosmos sink connector");
         this.sinkConfig = new CosmosSinkConfig(props);
         this.connectorName = props.containsKey(CONNECTOR_NAME) ? props.get(CONNECTOR_NAME).toString() : "EMPTY";
+        validateContainers();
+    }
+
+    private void validateContainers() {
```
[the remark this answers is not in the document]
Comment (tvaron3) [reply sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSinkConnector.java]: Created new utils class for this method and use it in sink and source connectors.

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSinkConnector.java [hunk 5]
```diff
@@ -41,6 +45,10 @@ public void start(Map<String, String> props) {
         LOGGER.info("Starting the kafka cosmos sink connector");
         this.sinkConfig = new CosmosSinkConfig(props);
         this.connectorName = props.containsKey(CONNECTOR_NAME) ? props.get(CONNECTOR_NAME).toString() : "EMPTY";
+        CosmosSinkContainersConfig containersConfig = this.sinkConfig.getContainersConfig();
+        CosmosAsyncClient cosmosAsyncClient = CosmosClientStore.getCosmosClient(this.sinkConfig.getAccountConfig(), this.connectorName);
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSinkConnector.java:49]: need to also close the clients after use

Comment (tvaron3) [reply sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSinkConnector.java:49]: closed the client.

**Pull request merged**